ambry-network/src/test/java/com/github/ambry/network/CompositeNetworkClientTest.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2020 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ */
+
+package com.github.ambry.network;
+
+import com.github.ambry.clustermap.DataNodeId;
+import com.github.ambry.clustermap.MockReplicaId;
+import com.github.ambry.clustermap.ReplicaType;
+import io.netty.buffer.Unpooled;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.After;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+
+/**
+ * Test {@link CompositeNetworkClient} and {@link CompositeNetworkClientFactory}.
+ */
+public class CompositeNetworkClientTest {
+  private final Mocks mockOne = new Mocks();
+  private final Mocks mockTwo = new Mocks();
```

Comment:
I had trouble following the mock usage.  Could we rename them diskMock and cloudMock to make it clearer?